src/protocols/dhcpv4/encode.c

```diff
@@ -73,28 +74,17 @@ static ssize_t encode_value(uint8_t *out, size_t outlen,
 	case FR_TYPE_ETHERNET:
 	case FR_TYPE_STRING:
 	case FR_TYPE_OCTETS:
-		len = fr_value_box_to_network(&need, p, outlen, &vp->data);
-		if (len < 0) return -2;
+		if (fr_value_box_to_network_dbuff(&need, dbuff, &vp->data) < 0) return -2;
```

Comment:
This return value is wrong, but it was always wrong.  Should return need as a negative integer, not -2.

Reply:
Corrected.